ipam/ipam.go

```diff
@@ -198,78 +218,87 @@ func (ipam *IPAM) allocateIP(input interface{}, ctx common.RestContext) (interfa
 		return nil, err
 	}
 	// Get host info from topology service
 	topoUrl, err := client.GetServiceUrl("topology")
 	if err != nil {
+		log.Printf("IPAM encountered an error getting a topology service URL %v", err)
 		return nil, err
 	}
 
 	index := common.IndexResponse{}
 	err = client.Get(topoUrl, &index)
 	if err != nil {
+		log.Printf("IPAM encountered an error querying topology: %v", err)
 		return nil, err
 	}
 
 	hostsURL := index.Links.FindByRel("host-list")
 	host := common.HostMessage{}
 
-	hostInfoURL := fmt.Sprintf("%s/%s", hostsURL, Endpoint.HostId)
+	hostInfoURL := fmt.Sprintf("%s/%s", hostsURL, endpoint.HostId)
 	err = client.Get(hostInfoURL, &host)
 
 	if err != nil {
+		log.Printf("IPAM encountered an error querying topology for hosts: %v", err)
 		return nil, err
 	}
 
 	tenantUrl, err := client.GetServiceUrl("tenant")
 	if err != nil {
+		log.Printf("IPAM encountered an error getting tenant srevice URL: %v", err)
 		return nil, err
 	}
 
 	// TODO follow links once tenant service supports it. For now...
 
 	t := &tenant.Tenant{}
-	tenantsUrl := fmt.Sprintf("%s/tenants/%s", tenantUrl, Endpoint.TenantID)
+	tenantsUrl := fmt.Sprintf("%s/tenants/%s", tenantUrl, endpoint.TenantID)
 	log.Printf("IPAM calling %s\n", tenantsUrl)
 	err = client.Get(tenantsUrl, t)
 	if err != nil {
+		log.Printf("IPAM encountered an error querying tenant service for tenant %s: %v", endpoint.TenantID, err)
 		return nil, err
 	}
 	log.Printf("IPAM: received tenant %s ID %d, sequence %d\n", t.Name, t.ID, t.Seq)
 
-	segmentUrl := fmt.Sprintf("/tenants/%s/segments/%s", Endpoint.TenantID, Endpoint.SegmentID)
+	segmentUrl := fmt.Sprintf("/tenants/%s/segments/%s", endpoint.TenantID, endpoint.SegmentID)
 	log.Printf("IPAM: calling %s\n", segmentUrl)
 	segment := &tenant.Segment{}
 	err = client.Get(segmentUrl, segment)
 	if err != nil {
+		log.Printf("IPAM encountered an error querying tenant service for tenant %s and segment %s: %v", endpoint.TenantID, endpoint.SegmentID, err)
 		return nil, err
 	}
 
 	log.Printf("Constructing IP from Host IP %s, Tenant %d, Segment %d", host.RomanaIp, t.Seq, segment.Seq)
 
-	EndpointBits := 32 - ipam.dc.PrefixBits - ipam.dc.PortBits - ipam.dc.TenantBits - ipam.dc.SegmentBits - ipam.dc.EndpointSpaceBits
-	segmentBitShift := EndpointBits
+	endpointBits := 32 - ipam.dc.PrefixBits - ipam.dc.PortBits - ipam.dc.TenantBits - ipam.dc.SegmentBits - ipam.dc.EndpointSpaceBits
+	segmentBitShift := endpointBits
 	//	prefixBitShift := 32 - ipam.dc.PrefixBits
 	tenantBitShift := segmentBitShift + ipam.dc.SegmentBits
 	log.Printf("Parsing Romana IP address of host %s: %s\n", host.Name, host.RomanaIp)
 	_, network, err := net.ParseCIDR(host.RomanaIp)
 	if err != nil {
+		log.Printf("IPAM encountered an error parsing %s: %v", host.RomanaIp, err)
 		return nil, err
 	}
 	hostIpInt := common.IPv4ToInt(network.IP)
-	tSeq := t.Seq - 1
-	sSeq := segment.Seq - 1
+	tSeq := t.Seq
```

Comment:
Since it doesn't do arithmetic on the value of `t.Seq` and `segment.Seq` anymore, it could just use those directly instead of creating `tSeq` and `sSeq`.